Sources/AsyncAlgorithms/Channels/ChannelStorage.swift

```diff
@@ -0,0 +1,149 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Async Algorithms open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+final class ChannelStorage<Element, Failure: Error>: @unchecked Sendable {
```

Comment:
From what I can tell this technically does not need to be a reference type; the `stateMachine` and `ids` storage is relegated to `ManagedCriticalState` which handles the mutations.

This can be addressed as a follow-up change

Reply:
done